### PR TITLE
sensor: dts - convert several sensor drivers to DTS

### PR DESCRIPTION
**Pull request by galak**
Convert th02 sensor driver and sample app to utilize device tree.

Introduce a dts board overlay on the frdm_k64f board to ensure we at
least have a single platform in which the sample gets built.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

### CODE REVIEW

File: dts/bindings/sensor/hoperf,th02.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Copyright (c) 2019, Linaro Limited
+# SPDX-License-Identifier: Apache-2.0
+
+title: TH02 Temperature and Humidity sensor
+
+description: |
+    This is a representation of TH02 Temperature and Humidity sensor
```
Comment (ulfalizer) [inline dts/bindings/sensor/hoperf,th02.yaml]: Could replace with

    TH02 Temperature and Humidity sensor. See datasheet at...

Or whatever other information might be useful.

Would be nice if no more "this is a (base) representation of..." could be added. :)

**Comment by ulfalizer** [comment]
> **Some checks failed. Please fix and resubmit.**
> 
> ## License issues
> In most cases you do not need to do anything here, especially if the files
> reported below are going into ext/ and if license was approved for inclusion
> into ext/ already. Fix any missing license/copyright issues. The license
> exception if a JFYI for the maintainers and can be overriden when merging the
> pull request.
> 
> * samples/sensor/th02/boards/frdm_k64f.overlay missing copyright.
> 
> _Tip: The bot edits this comment instead of posting a new one, so you can check the comment's history to see earlier messages._

Related PR: https://github.com/zephyrproject-rtos/ci-tools/pull/105

File: dts/bindings/sensor/hoperf,th02.yaml (hunk 1 above)
Comment (galak) [reply dts/bindings/sensor/hoperf,th02.yaml]: Fixed.

**Review by ulfalizer — APPROVED** [review]
LGTM

**Review by pabigot — COMMENTED** [review]
mcp9808 works (on particle_xenon, didn't test on k64f).  Can't test anything else.

(NB: The driver may other changes to work on Nordic, will submit those once this goes in.)

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: dts/bindings/sensor/semtech,sx9500.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+compatible: "semtech,sx9500"
+
+include: i2c-device.yaml
+
+properties:
+    int-gpios:
+      type: phandle-array
+      required: false
+
+    reset-gpios:
```
[the remark this answers is not in the document]
Comment (galak) [reply dts/bindings/sensor/semtech,sx9500.yaml]: future proofing the binding.  Happy to remove it.

File: dts/bindings/sensor/bosch,bmm150.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+description: |
+    Bosch BMM150 Geomagnetic sensor.  See more info at:
+    https://www.bosch-sensortec.com/bst/products/all_products/bmm150
+
+compatible: "bosch,bmm150"
+
+include: i2c-device.yaml
+
+properties:
+    int-gpios:
```
[the remark this answers is not in the document]
Comment (galak) [reply dts/bindings/sensor/bosch,bmm150.yaml]: future proofing the binding. Happy to remove it.

File: tests/drivers/build_all/dts_fixup.h [hunk 4]
```diff
@@ -297,6 +297,99 @@
 #define DT_TI_TMP116_BUS_I2C 1
 #endif
 
+#ifndef DT_INST_0_HOPERF_TH02_LABEL
```
Comment (carlescufi) [inline tests/drivers/build_all/dts_fixup.h]: Does it still make sense to keep adding those? It’s just for the non-DTS enabled platforms? Couldn’t the dt scripts generate them based on the overall DT files?

**Review by carlescufi — APPROVED** [review]
Looks good. I left a comment about the fixups but it’s unrelated to this particular PR.

File: tests/drivers/build_all/dts_fixup.h (hunk 4 above)
Comment (galak) [reply tests/drivers/build_all/dts_fixup.h]: It does for now, its the only way we have to ensure a build test for any driver that doesn't have an explicit sample.

The other option is to create a board overlay dts for this test that has all the sensors in it and that we'd keep adding as we have new sensor drivers.

Comment (galak) [reply tests/drivers/build_all/dts_fixup.h]: I do like that idea as it gives an example of what the .dts node should look like to someone wanting to add one of the sensors to their boards.

**Comment by galak** [comment]
@MaureenHelm I went ahead and remove the extra properties from the bindings since I had to rebase anyways.

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**